<table align="left" width=100%>
    <tr>
        <td width="20%">
            <img src="GL-2.png">
        </td>
        <td>
            <div align="center">
                <font color="#21618C" size=8px>
                  <b> Final Exam <br> </b>
                </font>
            </div>
        </td>
    </tr>
</table>

# NPV FINAL EXAM
## Total Marks 70

We combined 6 datasets in single CSV file and mention below columns belongs to

### 1. Report_2010_to_Congress_on_White_House_Staff


* Employee Name	
* Employee Status	
* Salary	
* Pay Basis	
* Position Title


### 2. chipotle


* order_id	
* quantity	
* item_name	
* choice_description	
* item_price


### 3. Euro_2012_stats_TEAM


* Team	
* Goals	
* Shots on target	
* Shots off target	
* Shooting Accuracy	
* % Goals-to-shots	
* Total shots (inc. Blocked)	
* Hit Woodwork	
* Penalty goals	
* Penalties not scored	
* Headed goals	
* Passes	
* Passes completed	
* Passing Accuracy	
* Touches	
* Crosses	
* Dribbles	
* Corners Taken	
* Tackles	
* Clearances	
* Interceptions	
* Clearances off line	
* Clean Sheets	
* Blocks	
* Goals conceded	
* Saves made	
* Saves-to-shots ratio	
* Fouls Won	
* Fouls Conceded	
* Offsides	
* Yellow Cards	
* Red Cards	
* Subs on	
* Subs off	
* Players Used


### 4. hotdogs


* Beef	
* 186	
* 495


### 5. stock_data

* Unknow column
* AA	
* AAPL	
* GE	
* IBM	
* JNJ	
* MSFT	
* PEP	
* SPX	
* XOM


### 6. Fifa

ID: unique id for every player

Name: name

Age: age

Nationality: nationality

Overall: overall rating

Potential: potential rating

Club: current club

Value: current market value

Wage: current wage

Special: special

Preferred Foot: left/right

International Reputation: rating on scale of 5

Weak Foot: rating on scale of 5

Skill Moves: rating on scale of 5

Work Rate: attack work rate/defense work rate

Body Type: body type of player

Real Face: true or false

Position: position on the pitch


Jersey Number: jersey number

Joined: joined date

Contract Valid Until: contract end date

Height: height of the player

Weight: weight of the player




## Kindly change the below cells from markdown to code and execute it 

* Note: Use encoding='latin1

In [6]:
%matplotlib inline
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
import math

In [12]:
import pandas as pd

# import csv

# with open("DATASET.csv","r")as file:

#     reader=csv.reader(file)

df=pd.read_csv("Data_Set.csv")

df.head()


Employee Name Employee Status  Salary  Pay Basis  \
0       Abrams, Adam W.        Employee   66300  Per Annum   
1         Adams, Ian H.        Employee   45000  Per Annum   
2       Agnew, David P.        Employee   93840  Per Annum   
3        Albino, James         Employee   91800  Per Annum   
4  Aldy, Jr., Joseph E.        Employee  130500  Per Annum   

                                      Position Title  order_id  quantity  \
0           WESTERN REGIONAL COMMUNICATIONS DIRECTOR         1         1   
1  EXECUTIVE ASSISTANT TO THE DIRECTOR OF SCHEDUL...         1         1   
2       DEPUTY DIRECTOR OF INTERGOVERNMENTAL AFFAIRS         1         1   
3                             SENIOR PROGRAM MANAGER         1         1   
4  SPECIAL ASSISTANT TO THE PRESIDENT FOR ENERGY ...         2         2   

                               item_name  \
0           Chips and Fresh Tomato Salsa   
1                                   Izze   
2                       Nantucket Nectar   
3  Chips and Tomatillo-Green Chili Salsa   
4                           Chicken Bowl   

                                  choice_description item_price  ...  \
0                                                NaN     $2.39   ...   
1                                       [Clementine]     $3.39   ...   
2                                            [Apple]     $3.39   ...   
3                                                NaN     $2.39   ...   
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   ...   

  Skill Moves       Work Rate  Body Type  Real Face Position Jersey Number  \
0           3  Medium/ Medium       Lean         No      RCM            10   
1           4    High/ Medium     Normal        Yes       LF             8   
2           4  Medium/ Medium     Stocky         No       CM             5   
3           3       Low/ High     Normal         No      LCM            11   
4           2    Medium/ High     Stocky         No      CDM             8   

      Joined  Contract Valid Until  Height  Weight  
0  19-Jul-14                  2019    5'10  143lbs  
1  16-Jul-18                  2021     5'7  150lbs  
2  09-Jul-15                  2019     5'7  159lbs  
3  31-Aug-17                  2019    5'11  168lbs  
4  06-Jul-18                  2019     5'6  154lbs  

[5 rows x 81 columns]

In [13]:
Report_2010_to_Congress_on_White_House_Staff = df.iloc[:, :5]
Report_2010_to_Congress_on_White_House_Staff.head(1)

Employee Name Employee Status  Salary  Pay Basis  \
0  Abrams, Adam W.        Employee   66300  Per Annum   

                             Position Title  
0  WESTERN REGIONAL COMMUNICATIONS DIRECTOR

In [14]:
chipotle = df.iloc[:, 5:10]
chipotle.head(1)

order_id  quantity                     item_name choice_description  \
0         1         1  Chips and Fresh Tomato Salsa                NaN   

  item_price  
0     $2.39

In [15]:
Euro_2012_stats_TEAM = df.iloc[:, 10:45]
Euro_2012_stats_TEAM.head(1)

Team  Goals  Shots on target  Shots off target Shooting Accuracy  \
0  Croatia      4               13                12            51.90%   

  % Goals-to-shots  Total shots (inc. Blocked)  Hit Woodwork  Penalty goals  \
0           16.00%                          32             0              0   

   Penalties not scored  ...  Saves made  Saves-to-shots ratio  Fouls Won  \
0                     0  ...          13                81.30%         41   

  Fouls Conceded  Offsides  Yellow Cards  Red Cards  Subs on  Subs off  \
0             62         2             9          0        9         9   

   Players Used  
0            16  

[1 rows x 35 columns]

In [16]:
hotdogs = df.iloc[:, 45:48]
hotdogs.head(1)

Beef  186  495
0  Beef  181  477

In [126]:
stock_data = df.iloc[:, 48:58]
stock_data.head(1)

Unnamed: 48     AA    AAPL     GE     IBM    JNJ   MSFT    PEP  \
0  29-10-2007 00:00  37.41  185.09  34.46  106.78  57.13  31.78  65.67   

       SPX    XOM  
0  1540.98  85.51

In [17]:
Fifa= df.iloc[:, 58:]
Fifa.head(1)

ID         Name  Age Nationality  Overall  Potential       Club  Value  \
0  16  Luis Garc¡a   37       Spain       71         71  KAS Eupen  ?750K   

  Wage  Special  ... Skill Moves       Work Rate  Body Type  Real Face  \
0  ?6K     1906  ...           3  Medium/ Medium       Lean         No   

  Position Jersey Number     Joined Contract Valid Until  Height  Weight  
0      RCM            10  19-Jul-14                 2019    5'10  143lbs  

[1 rows x 23 columns]

 _________________

# SECTION A: 20 MARKS

 _________________

### 1. Read the dataset(Report_2010_to_Congress_on_White_House_Staff) and Answer the below questions?
    
A. Find the salary of an employee in a (1st-row 2nd-column) using a position based indexing function? (1 Mark)

B. Display all the employees whose salary is greater than 50000?(1 Mark)

C. Display all the rows whose salary is greater than 50000 and position as "ANALYST"?(1 Mark)

D. Find the list of Employee details belonging to a position ('ANALYST','ADVISOR')?(1 Mark)

E. Find the top 10 highest-paid employees in each position?(1 Mark)


In [19]:
dfa = Report_2010_to_Congress_on_White_House_Staff
dfa.head()

Employee Name Employee Status  Salary  Pay Basis  \
0       Abrams, Adam W.        Employee   66300  Per Annum   
1         Adams, Ian H.        Employee   45000  Per Annum   
2       Agnew, David P.        Employee   93840  Per Annum   
3        Albino, James         Employee   91800  Per Annum   
4  Aldy, Jr., Joseph E.        Employee  130500  Per Annum   

                                      Position Title  
0           WESTERN REGIONAL COMMUNICATIONS DIRECTOR  
1  EXECUTIVE ASSISTANT TO THE DIRECTOR OF SCHEDUL...  
2       DEPUTY DIRECTOR OF INTERGOVERNMENTAL AFFAIRS  
3                             SENIOR PROGRAM MANAGER  
4  SPECIAL ASSISTANT TO THE PRESIDENT FOR ENERGY ...

In [23]:
dfa.iloc[1:2,2:3]

Salary
1   45000

In [25]:
dfa[dfa['Salary'] > 50000]

Employee Name Employee Status  Salary  Pay Basis  \
0               Abrams, Adam W.        Employee   66300  Per Annum   
2               Agnew, David P.        Employee   93840  Per Annum   
3                Albino, James         Employee   91800  Per Annum   
4          Aldy, Jr., Joseph E.        Employee  130500  Per Annum   
6          Amorsingh, Lucius L.        Employee   56092  Per Annum   
..                          ...             ...     ...        ...   
462          Wright, Jessica N.        Employee   96900  Per Annum   
463  Yeager Kaplan, Jennifer F.        Employee   66300  Per Annum   
464     Young, Jr., Reginald D.        Employee   60232  Per Annum   
466            Young, Latoya A.        Employee   58511  Per Annum   
468          Zichal, Heather R.        Employee  100000  Per Annum   

                                        Position Title  
0             WESTERN REGIONAL COMMUNICATIONS DIRECTOR  
2         DEPUTY DIRECTOR OF INTERGOVERNMENTAL AFFAIRS  
3                               SENIOR PROGRAM MANAGER  
4    SPECIAL ASSISTANT TO THE PRESIDENT FOR ENERGY ...  
6                                    SPECIAL ASSISTANT  
..                                                 ...  
462    DEPUTY DIRECTOR OF SCHEDULING FOR THE PRESIDENT  
463  SPECIAL ASSISTANT TO THE DIRECTOR OF PUBLIC EN...  
464                         RECORDS MANAGEMENT ANALYST  
466                         RECORDS MANAGEMENT ANALYST  
468  DEPUTY ASSISTANT TO THE PRESIDENT FOR ENERGY A...  

[321 rows x 5 columns]

In [28]:
dfa[(dfa['Position Title'] == 'ANALYST') & (dfa['Salary'] > 50000)]

Employee Name Employee Status  Salary  Pay Basis Position Title
453   White, Cleo        Detailee   65393  Per Annum        ANALYST

In [32]:
dfa[dfa['Position Title'].isin(['ANALYST','ADVISOR'])]


Employee Name Employee Status  Salary  Pay Basis Position Title
14      Ayling, Lindsay A.        Employee   42000  Per Annum        ANALYST
55        Campos, David R.        Employee   42000  Per Annum        ANALYST
63      Castillo, Laura M.        Employee   42000  Per Annum        ANALYST
104      Dorsett, Craig M.        Employee   42000  Per Annum        ANALYST
154      Grant, Kathryn A.        Employee   42000  Per Annum        ANALYST
167       Hegde, Shilpa S.        Employee   42000  Per Annum        ANALYST
277  McGinnis, Patricia G.        Employee       0  Per Annum        ADVISOR
284       Metcalf, Ryan D.        Employee   42000  Per Annum        ANALYST
338       Posey, Stacie M.        Employee   42000  Per Annum        ANALYST
353      Robertson, Ian T.        Employee   42000  Per Annum        ANALYST
373     Schaub, Matthew J.        Employee   42000  Per Annum        ANALYST
434       Utech, Daniel G.        Detailee  130000  Per Annum        ADVISOR
453            White, Cleo        Detailee   65393  Per Annum        ANALYST

In [36]:
dfa.groupby('Position Title')[['Employee Name','Salary']].max().head(10)

Employee Name  Salary
Position Title                                                  
ADVANCE COORDINATOR                  Faulman, Michael T.   42000
ADVISOR                                 Utech, Daniel G.  130000
ANALYST                                      White, Cleo   65393
ASSISTANT COUNSEL FOR NOMINATIONS       Kumar, Pravin D.   50000
ASSISTANT DIRECTOR                      Stracko, Jill A.   66300
ASSISTANT EXECUTIVE CLERK          Thomas, III, Edwin R.  106839
ASSISTANT PRESS SECRETARY              Vietor, Thomas F.   78000
ASSISTANT SHIFT LEADER                     Sass, Joan C.   63673
ASSISTANT SPEECHWRITER                 O'Connor, Kyle P.   42000
ASSISTANT STAFF SECRETARY                 Walsh, Joan L.   60000

### 2. Read the dataset(chipotle)

A. Select only the products with quantity equals to 1.  (1 marks)
 
B. Count the item_price is greater than 10(1 mark)
 
C. Find How many times Veggie Salad Bowl ordered? (1 mark)
 
D. Find the quantity of the most expensive item ordered? (2 mark)
 


In [65]:
dfb = chipotle
dfb.head()

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   

                                  choice_description item_price  
0                                                NaN     $2.39   
1                                       [Clementine]     $3.39   
2                                            [Apple]     $3.39   
3                                                NaN     $2.39   
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98

In [66]:
dfb = dfb.drop_duplicates(['item_name','quantity'])

In [67]:
dfb[dfb['quantity'] == 1]

order_id  quantity                              item_name  \
0           1         1           Chips and Fresh Tomato Salsa   
1           1         1                                   Izze   
2           1         1                       Nantucket Nectar   
3           1         1  Chips and Tomatillo-Green Chili Salsa   
5           3         1                           Chicken Bowl   
6           3         1                          Side of Chips   
7           4         1                          Steak Burrito   
8           4         1                       Steak Soft Tacos   
10          5         1                    Chips and Guacamole   
11          6         1                   Chicken Crispy Tacos   
12          6         1                     Chicken Soft Tacos   
16          8         1                        Chicken Burrito   
21         11         1                       Barbacoa Burrito   
27         14         1                       Carnitas Burrito   
28         14         1                            Canned Soda   
33         17         1                          Carnitas Bowl   
34         17         1                          Bottled Water   
38         18         1  Chips and Tomatillo Green Chili Salsa   
39         19         1                          Barbacoa Bowl   
40         19         1                                  Chips   
44         20         1                     Chicken Salad Bowl   
54         25         1                             Steak Bowl   
56         26         1                    Barbacoa Soft Tacos   
57         26         1                         Veggie Burrito   
62         28         1                            Veggie Bowl   
92         40         1                     Steak Crispy Tacos   
111        49         1    Chips and Tomatillo Red Chili Salsa   
168        75         1                  Barbacoa Crispy Tacos   
186        83         1                      Veggie Salad Bowl   
191        85         1     Chips and Roasted Chili-Corn Salsa   
233       102         1     Chips and Roasted Chili Corn Salsa   
237       103         1                    Carnitas Soft Tacos   
250       109         1                          Chicken Salad   
263       114         1                      Canned Soft Drink   
298       129         1                      6 Pack Soft Drink   
300       130         1    Chips and Tomatillo-Red Chili Salsa   

                                    choice_description item_price  
0                                                  NaN     $2.39   
1                                         [Clementine]     $3.39   
2                                              [Apple]     $3.39   
3                                                  NaN     $2.39   
5    [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...    $10.98   
6                                                  NaN     $1.69   
7    [Tomatillo Red Chili Salsa, [Fajita Vegetables...    $11.75   
8    [Tomatillo Green Chili Salsa, [Pinto Beans, Ch...     $9.25   
10                                                 NaN     $4.45   
11   [Roasted Chili Corn Salsa, [Fajita Vegetables,...     $8.75   
12   [Roasted Chili Corn Salsa, [Rice, Black Beans,...     $8.75   
16   [Tomatillo-Green Chili Salsa (Medium), [Pinto ...     $8.49   
21   [[Fresh Tomato Salsa (Mild), Tomatillo-Green C...     $8.99   
27   [[Tomatillo-Green Chili Salsa (Medium), Roaste...     $8.99   
28                                        [Dr. Pepper]     $1.09   
33   [Tomatillo-Red Chili Salsa (Hot), [Black Beans...     $8.99   
34                                                 NaN     $1.09   
38                                                 NaN     $2.95   
39   [Roasted Chili Corn Salsa, [Fajita Vegetables,...    $11.75   
40                                                 NaN     $2.15   
44   [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75   
54   [Fresh Tomato Salsa (Mild), [Black Beans, Rice...     $8.99  

In [70]:
def funa(x):
    return float(x[1:-1])

In [71]:
dfb['item_price'] = dfb['item_price'].apply(lambda x : funa(x))
dfb.head()

C:\Users\naras\AppData\Local\Temp\ipykernel_17804\967316975.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dfb['item_price'] = dfb['item_price'].apply(lambda x : funa(x))


order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   

                                  choice_description  item_price  
0                                                NaN        2.39  
1                                       [Clementine]        3.39  
2                                            [Apple]        3.39  
3                                                NaN        2.39  
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...       16.98

In [77]:
dfb[dfb['item_price'] > 10]['item_name'].value_counts().sum()

17

In [80]:
dfb[dfb['item_name'] == 'Veggie Salad Bowl'].value_counts().sum()

1

In [84]:
dfb[dfb['item_price'] ==(dfb['item_price'].max())]

order_id  quantity     item_name  \
409       178         3  Chicken Bowl   

                                    choice_description  item_price  
409  [[Fresh Tomato Salsa (Mild), Tomatillo-Green C...       32.94

### 3. Read the dataset(Euro_2012_stats_TEAM.CSV)

A. How many teams participated in the Euro2012 dataset.  (1 mark)

B. Create a data frame with a name "discipline" consisting of (Team, Yellow Cards and Red Cards) (1 mark)
C. Find average Yellow Cards given per Team (1 mark)

D. Find which teams  scored more than 6 goals(1 mark)

E. Display the teams start with " G "(1 mark)

In [86]:
dfc = Euro_2012_stats_TEAM
dfc.head()

Team  Goals  Shots on target  Shots off target Shooting Accuracy  \
0         Croatia      4               13                12            51.90%   
1  Czech Republic      4               13                18            41.90%   
2         Denmark      4               10                10            50.00%   
3         England      5               11                18            50.00%   
4          France      3               22                24            37.90%   

  % Goals-to-shots  Total shots (inc. Blocked)  Hit Woodwork  Penalty goals  \
0           16.00%                          32             0              0   
1           12.90%                          39             0              0   
2           20.00%                          27             1              0   
3           17.20%                          40             0              0   
4            6.50%                          65             1              0   

   Penalties not scored  ...  Saves made  Saves-to-shots ratio  Fouls Won  \
0                     0  ...          13                81.30%         41   
1                     0  ...           9                60.10%         53   
2                     0  ...          10                66.70%         25   
3                     0  ...          22                88.10%         43   
4                     0  ...           6                54.60%         36   

  Fouls Conceded  Offsides  Yellow Cards  Red Cards  Subs on  Subs off  \
0             62         2             9          0        9         9   
1             73         8             7          0       11        11   
2             38         8             4          0        7         7   
3             45         6             5          0       11        11   
4             51         5             6          0       11        11   

   Players Used  
0            16  
1            19  
2            15  
3            16  
4            19  

[5 rows x 35 columns]

In [90]:
dfc['Team'].nunique()

16

In [92]:
dfc1 = dfc[['Team','Yellow Cards','Red Cards']]
dfc1.groupby('Team')['Yellow Cards'].mean()

Team
Croatia                 9.0
Czech Republic          7.0
Denmark                 4.0
England                 5.0
France                  6.0
Germany                 4.0
Greece                  9.0
Italy                  16.0
Netherlands             5.0
Poland                  7.0
Portugal               12.0
Republic of Ireland     6.0
Russia                  6.0
Spain                  11.0
Sweden                  7.0
Ukraine                 5.0
Name: Yellow Cards, dtype: float64

In [94]:
dfc[dfc['Goals'] > 6].head()

Team  Goals  Shots on target  Shots off target Shooting Accuracy  \
5   Germany     10               32                32            47.80%   
13    Spain     12               42                33            55.90%   
21  Germany     10               32                32            47.80%   
29    Spain     12               42                33            55.90%   
37  Germany     10               32                32            47.80%   

   % Goals-to-shots  Total shots (inc. Blocked)  Hit Woodwork  Penalty goals  \
5            15.60%                          80             2              1   
13           16.00%                         100             0              1   
21           15.60%                          80             2              1   
29           16.00%                         100             0              1   
37           15.60%                          80             2              1   

    Penalties not scored  ...  Saves made  Saves-to-shots ratio  Fouls Won  \
5                      0  ...          10                62.60%         63   
13                     0  ...          15                93.80%        102   
21                     0  ...          10                62.60%         63   
29                     0  ...          15                93.80%        102   
37                     0  ...          10                62.60%         63   

   Fouls Conceded  Offsides  Yellow Cards  Red Cards  Subs on  Subs off  \
5              49        12             4          0       15        15   
13             83        19            11          0       17        17   
21             49        12             4          0       15        15   
29             83        19            11          0       17        17   
37             49        12             4          0       15        15   

    Players Used  
5             17  
13            18  
21            17  
29            18  
37            17  

[5 rows x 35 columns]

In [100]:
dfc[dfc['Team'].str.startswith('G')].head()

Team  Goals  Shots on target  Shots off target Shooting Accuracy  \
5   Germany     10               32                32            47.80%   
6    Greece      5                8                18            30.70%   
21  Germany     10               32                32            47.80%   
22   Greece      5                8                18            30.70%   
37  Germany     10               32                32            47.80%   

   % Goals-to-shots  Total shots (inc. Blocked)  Hit Woodwork  Penalty goals  \
5            15.60%                          80             2              1   
6            19.20%                          32             1              1   
21           15.60%                          80             2              1   
22           19.20%                          32             1              1   
37           15.60%                          80             2              1   

    Penalties not scored  ...  Saves made  Saves-to-shots ratio  Fouls Won  \
5                      0  ...          10                62.60%         63   
6                      1  ...          13                65.10%         67   
21                     0  ...          10                62.60%         63   
22                     1  ...          13                65.10%         67   
37                     0  ...          10                62.60%         63   

   Fouls Conceded  Offsides  Yellow Cards  Red Cards  Subs on  Subs off  \
5              49        12             4          0       15        15   
6              48        12             9          1       12        12   
21             49        12             4          0       15        15   
22             48        12             9          1       12        12   
37             49        12             4          0       15        15   

    Players Used  
5             17  
6             20  
21            17  
22            20  
37            17  

[5 rows x 35 columns]

In [101]:
dfc[dfc['Team'].str.endswith('y')].head()

Team  Goals  Shots on target  Shots off target Shooting Accuracy  \
5   Germany     10               32                32            47.80%   
7     Italy      6               34                45            43.00%   
21  Germany     10               32                32            47.80%   
23    Italy      6               34                45            43.00%   
37  Germany     10               32                32            47.80%   

   % Goals-to-shots  Total shots (inc. Blocked)  Hit Woodwork  Penalty goals  \
5            15.60%                          80             2              1   
7             7.50%                         110             2              0   
21           15.60%                          80             2              1   
23            7.50%                         110             2              0   
37           15.60%                          80             2              1   

    Penalties not scored  ...  Saves made  Saves-to-shots ratio  Fouls Won  \
5                      0  ...          10                62.60%         63   
7                      0  ...          20                74.10%        101   
21                     0  ...          10                62.60%         63   
23                     0  ...          20                74.10%        101   
37                     0  ...          10                62.60%         63   

   Fouls Conceded  Offsides  Yellow Cards  Red Cards  Subs on  Subs off  \
5              49        12             4          0       15        15   
7              89        16            16          0       18        18   
21             49        12             4          0       15        15   
23             89        16            16          0       18        18   
37             49        12             4          0       15        15   

    Players Used  
5             17  
7             19  
21            17  
23            19  
37            17  

[5 rows x 35 columns]

### 4.Read the dataset ( hotdog)

A. Give Column names as Type, Calories, Sodium? (1 Mark)

B. Find out average Calories and average Sodium vales for each Hotdog type (1 Mark)

C. Find out rows having Calories less than 120 and Sodium values greater than 300 (1 Marks)

D. Create Column Grade having values 'A' and 'B' as follows: 
    
Grade A:  Items having Calorie value less than 150 and Sodium value less than 400  

Grade B:   Otherwise (2 Marks)


In [104]:
dfd = hotdogs
dfd.head()

Beef  186  495
0  Beef  181  477
1  Beef  176  425
2  Beef  149  322
3  Beef  184  482
4  Beef  190  587

In [106]:
dfd.columns=['Type','Calories','Sodium']
dfd.head()

Type  Calories  Sodium
0  Beef       181     477
1  Beef       176     425
2  Beef       149     322
3  Beef       184     482
4  Beef       190     587

In [107]:
dfd.groupby('Type').mean()

Calories      Sodium
Type                           
Beef     155.915254  397.824859
Meat     158.705882  418.529412
Poultry  118.812950  457.733813

In [108]:
dfd[(dfd['Calories'] < 120) & (dfd['Sodium'] > 300)]

Type  Calories  Sodium
38   Poultry       102     396
39   Poultry       106     383
40   Poultry        94     387
41   Poultry       102     542
42   Poultry        87     359
..       ...       ...     ...
452  Poultry        87     359
453  Poultry        99     357
454  Poultry       107     528
455  Poultry       113     513
458  Poultry        86     358

[73 rows x 3 columns]

In [119]:
def func(x):
    if (x.Calories < 150) & (x.Sodium < 400) :
        return 'A'
    else:
        return 'B'

In [124]:
dfd['Group'] = dfd.apply(func,axis =1)
dfd

Type  Calories  Sodium Group
0    Beef       181     477     B
1    Beef       176     425     B
2    Beef       149     322     A
3    Beef       184     482     B
4    Beef       190     587     B
..    ...       ...     ...   ...
464  Beef       176     425     B
465  Beef       149     322     A
466  Beef       184     482     B
467  Beef       190     587     B
468  Beef       158     370     B

[469 rows x 4 columns]

--------------

# SECTION B: 20 MARKS

----------

### 5. Read the dataset (stock_data)

A. Give column names as `timestamp`, `AA`, `AAPL`, `GE`, `IBM`, `JNJ`, `MSFT`, `PEP`, `SPX`, `XOM`. List all the columns of the dataset, Identify the numerical, categorical columns. (4 Marks)

B. Convert the column `timestamp` into python datetime object (1 Marks)

C. Find Average value of stock `AAPL` in the year 2009  (5 Marks)


In [127]:
dfe = stock_data
dfe.head()

Unnamed: 48     AA    AAPL     GE     IBM    JNJ   MSFT    PEP  \
0  29-10-2007 00:00  37.41  185.09  34.46  106.78  57.13  31.78  65.67   
1  30-10-2007 00:00  36.43  187.00  34.39  106.15  56.99  32.70  65.80   
2  31-10-2007 00:00  36.79  189.95  34.97  108.01  57.30  33.84  65.69   
3  01-11-2007 00:00  35.22  187.44  34.27  105.72  56.85  34.07  64.51   
4  02-11-2007 00:00  35.83  187.87  34.27  106.59  56.95  34.07  65.03   

       SPX    XOM  
0  1540.98  85.51  
1  1531.02  83.25  
2  1549.38  84.03  
3  1508.44  80.84  
4  1509.65  80.32

In [130]:
dfe.rename(columns = {'Unnamed: 48' : 'timestamp'},inplace = True)
dfe

timestamp     AA    AAPL     GE     IBM    JNJ   MSFT    PEP  \
0    29-10-2007 00:00  37.41  185.09  34.46  106.78  57.13  31.78  65.67   
1    30-10-2007 00:00  36.43  187.00  34.39  106.15  56.99  32.70  65.80   
2    31-10-2007 00:00  36.79  189.95  34.97  108.01  57.30  33.84  65.69   
3    01-11-2007 00:00  35.22  187.44  34.27  105.72  56.85  34.07  64.51   
4    02-11-2007 00:00  35.83  187.87  34.27  106.59  56.95  34.07  65.03   
..                ...    ...     ...    ...     ...    ...    ...    ...   
464  01-09-2009 00:00  11.39  165.30  12.50  112.54  55.94  22.97  52.57   
465  02-09-2009 00:00  11.35  165.18  12.37  111.96  55.77  22.84  53.08   
466  03-09-2009 00:00  11.80  166.55  12.61  112.20  55.71  23.08  53.36   
467  04-09-2009 00:00  11.97  170.31  13.00  113.29  56.29  23.57  54.21   
468  08-09-2009 00:00  12.38  172.93  13.59  113.00  56.53  23.76  54.90   

         SPX    XOM  
0    1540.98  85.51  
1    1531.02  83.25  
2    1549.38  84.03  
3    1508.44  80.84  
4    1509.65  80.32  
..       ...    ...  
464   998.04  65.08  
465   994.75  64.86  
466  1003.24  64.93  
467  1016.40  65.81  
468  1025.39  67.21  

[469 rows x 10 columns]

In [133]:
dfe['timestamp'] = pd.to_datetime(dfe['timestamp'])
dfe

timestamp     AA    AAPL     GE     IBM    JNJ   MSFT    PEP      SPX  \
0   2007-10-29  37.41  185.09  34.46  106.78  57.13  31.78  65.67  1540.98   
1   2007-10-30  36.43  187.00  34.39  106.15  56.99  32.70  65.80  1531.02   
2   2007-10-31  36.79  189.95  34.97  108.01  57.30  33.84  65.69  1549.38   
3   2007-01-11  35.22  187.44  34.27  105.72  56.85  34.07  64.51  1508.44   
4   2007-02-11  35.83  187.87  34.27  106.59  56.95  34.07  65.03  1509.65   
..         ...    ...     ...    ...     ...    ...    ...    ...      ...   
464 2009-01-09  11.39  165.30  12.50  112.54  55.94  22.97  52.57   998.04   
465 2009-02-09  11.35  165.18  12.37  111.96  55.77  22.84  53.08   994.75   
466 2009-03-09  11.80  166.55  12.61  112.20  55.71  23.08  53.36  1003.24   
467 2009-04-09  11.97  170.31  13.00  113.29  56.29  23.57  54.21  1016.40   
468 2009-08-09  12.38  172.93  13.59  113.00  56.53  23.76  54.90  1025.39   

       XOM  
0    85.51  
1    83.25  
2    84.03  
3    80.84  
4    80.32  
..     ...  
464  65.08  
465  64.86  
466  64.93  
467  65.81  
468  67.21  

[469 rows x 10 columns]

In [137]:
dfe.groupby(dfe.timestamp.dt.year).mean().loc[2009]['AAPL']

124.87337209302326

### 6. Read the dataset(Report_2010_to_Congress_on_White_House_Staff) and Answer the below questions?
    
**Exercise:1** Find the position title to which more amount is spent on salaries? (6 marks)

**Exercise:2**

2A. Find the position title to which the least amount is spent on salaries? (2 marks)

2B) Find the highest-paid position title? (2 marks)

**Exercise:3**

3.A)Find the least paid position title? (1 marks)

3.B) Find the percentage of the amount spent on each position w.r.t Salaries? (3 marks)




In [139]:
dff = Report_2010_to_Congress_on_White_House_Staff
dff.head()

Employee Name Employee Status  Salary  Pay Basis  \
0       Abrams, Adam W.        Employee   66300  Per Annum   
1         Adams, Ian H.        Employee   45000  Per Annum   
2       Agnew, David P.        Employee   93840  Per Annum   
3        Albino, James         Employee   91800  Per Annum   
4  Aldy, Jr., Joseph E.        Employee  130500  Per Annum   

                                      Position Title  
0           WESTERN REGIONAL COMMUNICATIONS DIRECTOR  
1  EXECUTIVE ASSISTANT TO THE DIRECTOR OF SCHEDUL...  
2       DEPUTY DIRECTOR OF INTERGOVERNMENTAL AFFAIRS  
3                             SENIOR PROGRAM MANAGER  
4  SPECIAL ASSISTANT TO THE PRESIDENT FOR ENERGY ...

In [145]:
dff.groupby('Position Title').sum().max()

Salary    1305000
dtype: int64

In [146]:
dff.groupby('Position Title').sum().min()

Salary    42000
dtype: int64

In [168]:
o1 = dff.groupby('Position Title')['Salary'].sum().sort_values(ascending = False)
o1

Position Title
SPECIAL ASSISTANT TO THE PRESIDENT FOR LEGISLATIVE AFFAIRS                   1305000
SPECIAL ASSISTANT TO THE PRESIDENT AND ASSOCIATE COUNSEL TO THE PRESIDENT    1288500
STAFF ASSISTANT                                                               987840
RECORDS MANAGEMENT ANALYST                                                    795670
DEPUTY ASSOCIATE COUNSEL FOR PRESIDENTIAL PERSONNEL                           656211
                                                                              ...   
PRIORITY PLACEMENT STAFF ASSISTANT                                             42000
RESEARCH ASSOCIATE                                                             42000
BOARDS AND COMMISSIONS STAFF ASSISTANT                                         42000
DEPUTY ASSOCIATE DIRECTOR OF CORRESPONDENCE FOR THE FIRST LADY                 42000
ADVANCE COORDINATOR                                                            42000
Name: Salary, Length: 269, dtype: int64

In [167]:
dff.groupby('Position Title')['Salary'].sum().sort_values(ascending = True)

Position Title
ADVANCE COORDINATOR                                                            42000
LEGISLATIVE ASSISTANT AND ASSISTANT TO THE HOUSE LIAISON                       42000
ENERGY AND ENVIRONMENT STAFF ASSISTANT                                         42000
ECONOMICS STAFF ASSISTANT                                                      42000
RESEARCH ASSOCIATE                                                             42000
                                                                              ...   
DEPUTY ASSOCIATE COUNSEL FOR PRESIDENTIAL PERSONNEL                           656211
RECORDS MANAGEMENT ANALYST                                                    795670
STAFF ASSISTANT                                                               987840
SPECIAL ASSISTANT TO THE PRESIDENT AND ASSOCIATE COUNSEL TO THE PRESIDENT    1288500
SPECIAL ASSISTANT TO THE PRESIDENT FOR LEGISLATIVE AFFAIRS                   1305000
Name: Salary, Length: 269, dtype: int64

In [170]:
o2 = dff['Salary'].sum()
o2

38796307

In [171]:
result = o1 / o2 * 100
result

Position Title
SPECIAL ASSISTANT TO THE PRESIDENT FOR LEGISLATIVE AFFAIRS                   3.363722
SPECIAL ASSISTANT TO THE PRESIDENT AND ASSOCIATE COUNSEL TO THE PRESIDENT    3.321192
STAFF ASSISTANT                                                              2.546222
RECORDS MANAGEMENT ANALYST                                                   2.050891
DEPUTY ASSOCIATE COUNSEL FOR PRESIDENTIAL PERSONNEL                          1.691426
                                                                               ...   
PRIORITY PLACEMENT STAFF ASSISTANT                                           0.108258
RESEARCH ASSOCIATE                                                           0.108258
BOARDS AND COMMISSIONS STAFF ASSISTANT                                       0.108258
DEPUTY ASSOCIATE DIRECTOR OF CORRESPONDENCE FOR THE FIRST LADY               0.108258
ADVANCE COORDINATOR                                                          0.108258
Name: Salary, Length: 269, dtype: float

------------------
# SECTION C: 30 MARKS

------------------

### 7. Read the dataset(Fifa) and Answer the below questions?

In [172]:
dfg = Fifa

In [173]:
dfg.head()

ID          Name  Age Nationality  Overall  Potential  \
0   16   Luis Garc¡a   37       Spain       71         71   
1   41       Iniesta   34       Spain       86         86   
2   80  E. Bel”zo?lu   37      Turkey       79         79   
3  164      G. Pinzi   37       Italy       70         70   
4  657    D. Vaughan   35       Wales       66         66   

                    Club   Value  Wage  Special  ... Skill Moves  \
0              KAS Eupen   ?750K   ?6K     1906  ...           3   
1            Vissel Kobe  ?21.5M  ?21K     2058  ...           4   
2  Medipol Ba?ak?ehir FK     ?4M  ?23K     2047  ...           4   
3                 Padova   ?240K   ?2K     1882  ...           3   
4           Notts County   ?150K   ?4K     1781  ...           2   

        Work Rate  Body Type  Real Face Position Jersey Number     Joined  \
0  Medium/ Medium       Lean         No      RCM            10  19-Jul-14   
1    High/ Medium     Normal        Yes       LF             8  16-Jul-18   
2  Medium/ Medium     Stocky         No       CM             5  09-Jul-15   
3       Low/ High     Normal         No      LCM            11  31-Aug-17   
4    Medium/ High     Stocky         No      CDM             8  06-Jul-18   

  Contract Valid Until  Height  Weight  
0                 2019    5'10  143lbs  
1                 2021     5'7  150lbs  
2                 2019     5'7  159lbs  
3                 2019    5'11  168lbs  
4                 2019     5'6  154lbs  

[5 rows x 23 columns]

### 7A).	Data Manipulation for Value Column (5 Marks).

**Step1:** Clean Value column by removing '?' from each row.

**Step2:** Convert the column in uniform unit i.e. 'K' by using given conversion rate 1 =1000 K and also remove the letter 'K' and 'M' from each row.

**Step3:** Convert the Datatype of the column to 'float'

**First 5 values of Sample output column should look like this:**


|0   |    750.0|
|----|-------|
|1   |   21500.0|
|2    |   4000.0|
|3    |   240.0|
|4    |   150.0|

Name: Value, dtype: float64



In [195]:
a= Fifa.describe()['Age']['max']
a

44.0

### 7B).	Data Manipulation for Height Column (5 Marks).

Step1: Convert Height column from Feet’ Inches unit to Inches unit by using given conversion rate 1 feet = 12 inches
Step2: Convert the Datatype of the Height column to 'int'

First 5 values of Sample Output Height column should look like this:
    
|0 |   70|
|-|-|
|1|    67|
|2 |   67|
|3  |  71|
|4 |   66|
Name: Height, dtype: int32


### 7C).	Data Manipulation for Weight Column (5 Marks).

**Step1:** Clean Weight column by removing ‘lbs’ from each row’

**Step2:** Convert the Datatype of the Weight column to 'int'

**First 5 values of Sample Output Weight column should look like this:**

|0  |  143|
|--|--|
|1  |  150|
|2  |  159|
|3  |  168|
|4  |  154|
|Name: Weight,| dtype: int32|


### 7D).   Find how many different countries players data is included in the dataset (1 Mark)

### 7E).	Find out highest Valued player from the dataset (1 Mark)

### 7F).	Find which country's players data is maximum in the dataset and what’s is the count of it (2 Marks)

### 7G)    Display top 10 clubs ordered by sum of all player’s current market value of that club (3 Marks)


In [191]:
Fifa.groupby('Club')['Value'].sum()

Club
1. FC Kaiserslautern              ?250K
1. FC K”ln                        ?725K
1. FSV Mainz 05                   ?6.5M
AC Ajaccio                        ?800K
AC Horsens                    ?70K?170K
                             ...       
Wis?a Krak¢w                 ?300K?240K
Wolfsberger AC                    ?1.3M
Wycombe Wanderers       ?180K?150K?100K
Yeovil Town                    ? 0?240K
€aykur Rizespor                   ?525K
Name: Value, Length: 306, dtype: object

### 7H).    Count the number of players having International Reputation ratings = 4 from the clubs 'Juventus’ and ‘Be?ikta? JK’ (3 Marks)

### 7I).	Find out the names of the youngest and oldest players from the dataset and display the distribution of age using DISTPLOT (5 Marks)